stdlib/public/core/UnicodeSPI.swift

```diff
@@ -199,3 +199,32 @@ extension Unicode.Scalar.Properties {
     return result
   }
 }
+
+//===----------------------------------------------------------------------===//
+// String Word Breaking
+//===----------------------------------------------------------------------===//
+
+extension String {
+  @_spi(_Unicode)
+  @available(SwiftStdlib 5.7, *)
+  public func _wordIndex(after i: String.Index) -> String.Index {
+    _precondition(i < endIndex)
+
+    let i = _guts.validateWordIndex(i)
+
+    let next = _guts.nextWordIndex(startingAt: i._encodedOffset)
+    return String.Index(_encodedOffset: next)
+  }
+
+  @_spi(_Unicode)
+  @available(SwiftStdlib 5.7, *)
+  public func _wordIndex(before i: String.Index) -> String.Index {
+    _precondition(i > startIndex)
+    _precondition(i <= endIndex)
+
+    let i = _guts.validateInclusiveWordIndex(i)
```

Comment:
This isn't right. `validateInclusiveWordIndex` can round the index down to the `startIndex`, so the `i > startIndex` check must come after the call. The `i <= endIndex` check is superfluous, as the validation call does that far better.

```suggestion
    let i = _guts.validateInclusiveWordIndex(i)
    _precondition(i > startIndex, "String index is out of bounds")
```